team/bundles/org.eclipse.compare/compare/org/eclipse/compare/contentmergeviewer/ContentMergeViewer.java

```diff
@@ -797,8 +805,12 @@ private void internalRefresh(Object input) {
 				ToolBarManager tbm = (ToolBarManager) getToolBarManager(fComposite.getParent());
 				if (tbm != null ) {
 					updateToolItems();
-					tbm.update(true);
-					tbm.getControl().getParent().layout(true);
+					Display.getDefault().asyncExec(() -> {
```

Comment:
Key change, run pack and re-layout in next tick

Reply:
why was this changed from `layout()` to `setRedraw()`?

Also, since we are now in an `asyncExec()`, the control might have been disposed in the mean time.

This needs ta `tbm.getControl().isDisposed()` check first thing in the lambda.

Reply:
@zulus: Any update on this?
We are seeing test failures due to this:
```
java.lang.NullPointerException: Cannot invoke "org.eclipse.swt.widgets.ToolBar.getParent()" because the return value of "org.eclipse.jface.action.ToolBarManager.getControl()" is null
 at org.eclipse.compare.contentmergeviewer.ContentMergeViewer.lambda$5(ContentMergeViewer.java:811)
 at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
 at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:132)
 at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:5040)
 at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4520)
 at org.eclipse.jface.operation.ModalContext$ModalContextThread.block(ModalContext.java:166)
 at org.eclipse.jface.operation.ModalContext.run(ModalContext.java:368)
 at org.eclipse.jface.dialogs.ProgressMonitorDialog.run(ProgressMonitorDialog.java:470)
 at org.eclipse.ui.internal.progress.ProgressMonitorJobsDialog.run(ProgressMonitorJobsDialog.java:230)
 at org.eclipse.ui.internal.progress.ProgressManager.lambda$26(ProgressManager.java:836)
 at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:67)
 at org.eclipse.ui.internal.progress.ProgressManager.busyCursorWhile(ProgressManager.java:869)
 at org.eclipse.ui.internal.progress.ProgressManager.busyCursorWhile(ProgressManager.java:845)
 at org.eclipse.ui.internal.progress.ProgressManager.run(ProgressManager.java:1003)
 at org.eclipse.compare.internal.Utilities.executeRunnable(Utilities.java:911)
 at org.eclipse.compare.internal.Utilities.executeRunnable(Utilities.java:894)
 at org.eclipse.compare.internal.CompareUIPlugin.compareResultOK(CompareUIPlugin.java:621)
 at org.eclipse.compare.internal.CompareUIPlugin.openCompareEditor(CompareUIPlugin.java:539)
 at org.eclipse.compare.CompareUI.openCompareEditorOnPage(CompareUI.java:174)
 at org.eclipse.compare.CompareUI.openCompareEditor(CompareUI.java:143)
 at org.eclipse.compare.CompareUI.openCompareEditor(CompareUI.java:127)
```

And why was
```
tbm.getControl().getParent().layout(true);
```
changed to
```
tbm.getControl().getParent().setRedraw(true);
```
?
Was this by accident or an intended change?